EventFilter/L1TRawToDigi/plugins/implementations_stage2/ZDCUnpacker.cc

```diff
@@ -0,0 +1,71 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "EventFilter/L1TRawToDigi/plugins/UnpackerFactory.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1TObjectCollections.h"
+
+#include "L1TStage2Layer2Constants.h"
+#include "ZDCUnpacker.h"
+
+namespace l1t {
+  namespace stage2 {
+    ZDCUnpacker::ZDCUnpacker() : ZDCSumCopy_(0) {}
+
+    bool ZDCUnpacker::unpack(const Block& block, UnpackerCollections* coll) {
+      using namespace l1t::stage2::layer2;

```

Comment:
this should just be `l1t::stage2`, since the ZDC is not part of Calo Layer 2

Reply:
I addressed this in my commit, Fixing Issues for PR 42733 (First Attempt). I had to change demux::nOutputFramePerBX to layer2::demux::nOutputFramePerBX to solve compiler errors. The ZDC data has 6 frames for BX which is the value of nOutputFramePerBX but this may be coincident.

Reply:
Yeah, I was unsure whether to flag the use of `demux::nOutputFramePerBX` in my first pass. I'd say this could technically change and become inconsistent with the number of input frames to the uGT, but in practice I have a hard time imagining how that would do. (So the bottom line from my side was that it's probably reasonable. If I remember correctly we just put the literal 6 in the muon unpacker which is also not nice.. ) @bundocka may have better thought through input than me, though.

Reply:
demux::nOutputFramePerBX == zdc::nOutputFramePerBX is a coincidence, I would not recommend using the demux constant, but instead create zdc::nOutputFramePerBX, for clarity. as Dinyar said, this should be fine as it is, and it shouldn't change, but the two numbers are technically unrelated so they shouldn't use the same constant.

Reply:
I tell a lie, it is not a coincidence at all :)  6bx is the maximum length allowed for the GT input packet. the input links are clocked at 240 MHz. 240 / 6 = 40 MHz.

Reply:
To add the zdc::nOutputFramePerBX variable. should I add it to the L1TStage2Layer2Constants.cc file or create a new constants file only for zdc.  I could also add it to the ZDCUnpacker.h but I don't like that since I believe the variable will also be used by the zdc packer in the future. 

 I would have the variable be this in all cases

 const unsigned int l1t::stage2::zdc::nOutputFramePerBX = 6;

Reply:
I would be tempted to put it somewhere like  GTSetup.h, since it is a general parameter for the input to the GT. (Then technically this parameter should be used throughout the muon and layer 2 unpackers)
https://github.com/cms-sw/cmssw/blob/master/EventFilter/L1TRawToDigi/plugins/implementations_stage2/GTSetup.h